llvm/lib/Target/NVPTX/NVPTXISelDAGToDAG.cpp

```diff
@@ -820,68 +845,76 @@ static unsigned int getCodeMemorySemantic(MemSDNode *N,
       (CodeAddrSpace == NVPTX::PTXLdStInstCode::GENERIC ||
        CodeAddrSpace == NVPTX::PTXLdStInstCode::GLOBAL ||
        CodeAddrSpace == NVPTX::PTXLdStInstCode::SHARED);
+  if (!AddrGenericOrGlobalOrShared)
+    return NVPTX::Ordering::NotAtomic;
```

Comment:
Are there any cases we may want to use volatile here? 

We probably do not need it for 'local'. But I'm not quite sure about `.const`. Technically it could be updated by the host and we may need to have a way to force a read from it.

Reply:
Right now, according to the PTX spec, doing a `volatile` of a `.const` is undefined behavior. 
We should probably fix that in the spec, the PTX implementation, and once we add the `.const` and `.param` tests, update all of this as needed.

Reply:
> Technically it could be updated by the host and we may need to have a way to force a read from it.

If the host updates a `const` while a kernel that has accessed it once has not finished, the behavior is undefined.